### PR TITLE
[Guide] Add guide to set up CLion for compiler development

### PR DESCRIPTION
**Pull request by ktoso**
Added information how to set up CLion for compiler development.

Adding upon request from @shahmishal, since some other people may be wanting to work on Swift on platforms where there is no Xcode etc. :-)

### CODE REVIEW

File: docs/HowToGuides/GettingStarted.md [hunk 1]
```diff
@@ -358,6 +359,28 @@ select the following schemes:
   run.
 <!-- TODO: Insert SourceKit/stdlib specific instructions? -->
 
+### Other IDEs setup
+
+You can also use other editors and IDEs to work on Swift.
+
+#### IntelliJ CLion
+
+CLion supports CMake and Ninja. In order to configure it property, build the swift project first using the `build-script`, then open the `swift` directory with CLion and proceed to project settings (`cmd + ,`).
```
Comment (buttaface) [inline docs/HowToGuides/GettingStarted.md:368]: property -> properly?

File: docs/HowToGuides/GettingStarted.md [hunk 2]
```diff
@@ -358,6 +359,28 @@ select the following schemes:
...
+- Build type: This corresponds to `CMAKE_BUILD_TYPE` so should be e.g. `RelWithDebInfoAssert` or `Debug`
+    - latest versions of the IDE suggest valid values here. Generally `RelWithDebInfoAssert` is a good one to work with
+- Toolchain: Default should be fine
+- Generator: Ninja
+- CMake options:
+    - `-D SWIFT_PATH_TO_CMARK_BUILD=SOME_PATH/swift-project/build/Ninja-RelWithDebInfoAssert/cmark-macosx-arm64 -D LLVM_DIR=SOME_PATH/swift-project/build/Ninja-RelWithDebInfoAssert/llvm-macosx-arm64/lib/cmake/llvm -D Clang_DIR=SOME_PATH/swift-project/build/Ninja-RelWithDebInfoAssert/llvm-macosx-arm64/lib/cmake/clang -D CMAKE_BUILD_TYPE=RelWithDebInfo -G Ninja -S .`
+    - replace the `SOME_PATH` to the path where your `swift-project` directory is
+    - the CMAKE_BUILD_TYPE should match the build configuration name, so if you named this profile `RelWithDebInfo` the CMAKE_BUILD_TYPE should also be `RelWithDebInfo`
+
+With this done CLion should be able to successfully import the project and have full autocomplete and code navigation powers.
```
Comment (buttaface) [inline docs/HowToGuides/GettingStarted.md:382]: done,

File: docs/HowToGuides/GettingStarted.md (hunk 1 above)
Comment (ktoso) [reply docs/HowToGuides/GettingStarted.md:368]:
```suggestion
CLion supports CMake and Ninja. In order to configure it properly, build the swift project first using the `build-script`, then open the `swift` directory with CLion and proceed to project settings (`cmd + ,`).
```

File: docs/HowToGuides/GettingStarted.md (hunk 2 above)
Comment (ktoso) [reply docs/HowToGuides/GettingStarted.md:382]:
```suggestion
With this done, CLion should be able to successfully import the project and have full autocomplete and code navigation powers.
```

**Comment by ktoso** [comment]
Thanks

**Comment by ktoso** [comment]
@swift-ci please smoke test

File: docs/HowToGuides/GettingStarted.md [hunk 3]
```diff
@@ -358,6 +359,28 @@ select the following schemes:
...
+
+- Name: mirror the name of the build configuration here, e.g. `RelWithDebInfoAssert` or `Debug`
+- Build type: This corresponds to `CMAKE_BUILD_TYPE` so should be e.g. `RelWithDebInfoAssert` or `Debug`
+    - latest versions of the IDE suggest valid values here. Generally `RelWithDebInfoAssert` is a good one to work with
+- Toolchain: Default should be fine
+- Generator: Ninja
+- CMake options:
+    - `-D SWIFT_PATH_TO_CMARK_BUILD=SOME_PATH/swift-project/build/Ninja-RelWithDebInfoAssert/cmark-macosx-arm64 -D LLVM_DIR=SOME_PATH/swift-project/build/Ninja-RelWithDebInfoAssert/llvm-macosx-arm64/lib/cmake/llvm -D Clang_DIR=SOME_PATH/swift-project/build/Ninja-RelWithDebInfoAssert/llvm-macosx-arm64/lib/cmake/clang -D CMAKE_BUILD_TYPE=RelWithDebInfo -G Ninja -S .`
+    - replace the `SOME_PATH` to the path where your `swift-project` directory is
+    - the CMAKE_BUILD_TYPE should match the build configuration name, so if you named this profile `RelWithDebInfo` the CMAKE_BUILD_TYPE should also be `RelWithDebInfo`
```
Comment (WowbaggersLiquidLunch) [inline docs/HowToGuides/GettingStarted.md:380]: I think the guide should use either `RelWithDebInfo` or `RelWithDebInfoAssert` consistently as `CMAKE_BUILD_TYPE`.

**Comment by WowbaggersLiquidLunch** [comment]
The way I use CLion with this project is building the compiler using Ninja first, then open the build directory directly in CLion. CMake options are automatically set to only `-G "Ninja"`, which is much fewer than what's suggested in the guide. With my limited experience, this works fine. Can this be added to the guide as an alternative approach to first-time setup?

File: docs/HowToGuides/GettingStarted.md (hunk 3 above)
Comment (ktoso) [reply docs/HowToGuides/GettingStarted.md:380]: sure, let's stick with RelWithDebInfoAssert

**Comment by ktoso** [comment]
I got to this working a while ago and never poked at it anymore since -- feel free to edit the guide and suggest changes! 👍

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Pull request merged**